src/core/lwt_pool.mli

```diff
@@ -27,31 +27,45 @@
     The pool also provides a limit on the number of connections that can
     simultaneously be open.
 
-    Note that pools are not for keeping Lwt threads. Lwt threads are very cheap
-    to create and are pure. It is neither desirable nor possible to reuse them.
-    If you want to have a pool of {e system} threads, consider module
+    Note that pools are not for keeping Lwt promises. Lwt promises are very
+    cheap to create. It is neither desirable nor possible to reuse them.  If
+    you want to have a pool of {e system} threads, consider using
     [Lwt_preemptive]. *)
 
 type 'a t
-  (** Pools containing values of type ['a]. *)
+  (** Pools containing elements of type ['a]. *)
 
 val create :
   int ->
   ?check : ('a -> (bool -> unit) -> unit) ->
   ?validate : ('a -> bool Lwt.t) ->
```

Comment:
Optional, but maybe we should also change the order so that `?validate` comes before `?check`? On each call to `use`, `?validate` is called first, *then* `?check` if the promise returned by `?use` is rejected.